src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs

```diff
@@ -13,9 +15,20 @@ internal class KubernetesEventGenerator : LinuxEventGenerator
         private const int MaxDetailsLength = 10000;
         private readonly Action<string> _writeEvent;
 
-        public KubernetesEventGenerator(Action<string> writeEvent = null)
+        public KubernetesEventGenerator(IOptions<ConsoleLoggingOptions> consoleLoggingOptions, Action<string> writeEvent = null)
```

Comment:
Why not just remove the writeEvent Action and use the same pattern that LinuxContainerEventGenerator does, where it uses ConsoleLoggingOptions.Writer which defaults to Console.Out, and also allows for tests to override (see the tests for LinuxContainerEventGenerator).

This'll make these consistent, and simplify your logic below.